mars/services/task/execution/ray/executor.py

```diff
@@ -154,11 +173,19 @@ def __init__(
 
         self._available_band_resources = None
 
-        # For progress
+        # For progress and task cancel
         self._pre_all_stages_progress = 0.0
-        self._pre_all_stages_tile_progress = 0
-        self._cur_stage_tile_progress = 0
+        self._pre_all_stages_tile_progress = 0.0
+        self._cur_stage_progress = 0.0
+        self._cur_stage_tile_progress = 0.0
+        self._cur_stage_subtask_to_output_object_refs = dict()
         self._cur_stage_output_object_refs = []
+        self._cur_stage_unfinished_subtasks = set()
+        self._cur_stage_submitted_subtasks = set()
+        self._execute_subtask_graph_aiotask = None
+        self._done = asyncio.Event()
```

Comment:
It's error prone to remove the event before set it because some logic may waiting for this event.

Reply:
Removed it. Thanks.